File: client/src/test/scala/stasis/test/specs/unit/client/mocks/MockBackupCollector.scala
package stasis.test.specs.unit.client.mocks

import org.apache.pekko.NotUsed
import org.apache.pekko.stream.scaladsl.Source
import stasis.client.collection.BackupCollector
import stasis.client.model.SourceEntity

class MockBackupCollector(files: List[SourceEntity]) extends BackupCollector {
  override def collect(): Source[SourceEntity, NotUsed] = Source(files)
}

File: client-android/lib/src/test/kotlin/stasis/test/client_android/lib/mocks/MockBackupCollector.kt
package stasis.test.client_android.lib.mocks

import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.asFlow
import stasis.client_android.lib.collection.BackupCollector
import stasis.client_android.lib.model.SourceEntity

class MockBackupCollector(val files: List<SourceEntity>) : BackupCollector {
    override fun collect(): Flow<SourceEntity> = files.asFlow()
}
